### PR TITLE
Update snow node ami builder doc

### LINKED ISSUE #5706: Improve Snow node image builder doc
https://anywhere.eks.amazonaws.com/docs/reference/artifacts/#build-snow-node-images

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

Resolves #5706 

*Description of changes:*

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/reference/artifacts.md [hunk 1]
```diff
@@ -747,27 +747,69 @@ These steps use `image-builder` to create an Ubuntu-based Amazon Machine Image (
...
+   ##### **aws_region**
+   The AWS region in which to launch the EC2 instance to create the AMI. (default: `us-west-2`).
+
+   ##### **ansible_extra_vars**
+   The absolute path to the additional variables to pass to Ansible. These are converted to the `--extra-vars` command-line argument. This path must be prefix with '@'. (default: `@/home/image-builder/eks-anywhere-build-tooling/projects/kubernetes-sigs/image-builder/packer/ami/ansible_extra_vars.yaml`)
+
+   ##### **builder_instance_type**
+   The EC2 instance type to use while building the AMI. (default: `t3.small`).
+
+   ##### **custom_role**
```
Comment (jiayiwang7) [inline docs/content/en/docs/reference/artifacts.md:789]: @abhay-krishna does this need to be true for all? Base on Nathan's comment:
> It seems like this must be set to "true" in order to load the bootstrap.sh and logging.sh files into the EKS-D Node AMIs.

File: docs/content/en/docs/reference/artifacts.md [hunk 2]
```diff
@@ -747,27 +747,69 @@ These steps use `image-builder` to create an Ubuntu-based Amazon Machine Image (
...
-     "custom_role_name_list" : "<Array of strings representing the absolute paths of custom Ansible roles to run. This field is mutually exclusive with custom_role_names>",
-     "custom_role_names": "<Space-delimited string of the custom roles to run. This field is mutually exclusive with custom_role_name_list and is provided for compatibility with Ansible's input format>",
-     "manifest_output": "<The absolute path to write the build artifacts manifest to. If you wish to export the AMI using this manifest, ensure that you provide a path that is not inside the '/home/$USER/eks-anywhere-build-tooling' path since that will be cleaned up when the build finishes>",
-     "root_device_name": "<The device name used by EC2 for the root EBS volume attached to the instance>",
-     "subnet_id": "<The ID of the subnet where Packer will launch the EC2 instance. This field is required when using a non-default VPC>",
-     "volume_size": "<The size of the root EBS volume in GiB>",
-     "volume_type": "<The type of root EBS volume, such as gp2, gp3, io1, etc.>"
+      "ami_filter_name": "ubuntu/images/*ubuntu-focal-20.04-amd64-server-*",
+      "ami_filter_owners": "amazon",
```
Comment (yxinchen) [inline docs/content/en/docs/reference/artifacts.md:757]: We should use the default value `679593333241` or `aws-marketplace` here. Marketplace AMI will be listed in customer's resource by default, but the amazon one needs an additional allowlist step, which is not recommended by Snow.

File: docs/content/en/docs/reference/artifacts.md (hunk 1 above)
Comment (abhay-krishna) [reply docs/content/en/docs/reference/artifacts.md:789]: Yes it needs to be true. We can remove this param so that customers don't set it to false

**Review by abhay-krishna — APPROVED** [review]
/lgtm
/approve

**Pull request merged**